view/frontend/web/js/view/payment/method-renderer/omise-offsite-installment-method.js

```diff
@@ -56,19 +58,120 @@ define(
+            getFormattedAmount: function (amount) {
+                return priceUtils.formatPrice(amount, quote.getPriceFormat());
+            },
+
+            /**
+             * Get Installment minimum
+             * this function respects info from: https://www.omise.co/installment-payment
+             *
+             * NOTE: in the future this function should return data from capabilities object.
+             *
+             * @param {string} id - Bank ID
+             * @return {integer}
+             */
+            getInstallmentMinimum: function (id) {
+                switch (id) {
+                    case 'kbank': return 500;
+                    case 'bbl': return 500;
+                    case 'bay': return 300;
+                    case 'first_choice': return 300;
+                    case 'ktc': return 300;
+                }
+            },
+
+            /**
+             * Get Installment monthly interest rate
+             *
+             * NOTE: in the future this function should return data from capabilities object.
+             * 
+             * @param {string} id - Bank id
+             * @return {float}
+             */
+            getInstallmentInterestRate: function (id) {
+                switch (id) {
+                    case 'kbank': return 0.0065;
+                    case 'bbl': return 0.008;
+                    case 'bay': return 0.008;
+                    case 'first_choice': return 0.013;
+                    case 'ktc': return 0.008;
+                }
+            },
+
+            /**
+             * Get VAT tax rate
+             *
+             * NOTE: in the future this function should return data from capabilities object.
+             *
+             * @return {float}
+             */
+            getVatRate: function () {
+                return 0.07;
+            },
+
+            /**
+             * Get zero interest setting
+             *
+             * @return {string}
+             */
+            isZeroInterest: function () {
+                return window.checkoutConfig.is_zero_interest;
+            },
+
+            /**
+             * Calculates single installment amount
+             * 
+             * @param {string} id - Bank ID
+             * @param {integer} terms - number of monthly installments
+             * @return {integer}
+             */
+            calculateSingleInstallmentAmount: function (id, terms) {
+                const total = this.getTotal();
+                const tax = this.getVatRate();
+
+                if (this.isZeroInterest()) {
+                    //merchant pays interest
+                    return (total / terms).toFixed(0)
+                }
+
+                // get monthly interest setting
+                const rate = this.getInstallmentInterestRate(id);
+
+                //calculate interests
+                const interest = ((1 + tax) * rate * terms * total);
+
+                return + (((total + interest) / terms).toFixed(0));
```

Comment:
About `toFixed(0)`, can you help confirm with the team if it's ok to round up those decimals? I'm not sure if it's the best way to cut all decimals out.

Reply:
I am fine with the current solution.

Reply:
changed to 2 decimals @guzzilar 
i was big fan of not showing decimals but now magento framework is showing those decimals, so it would show XXXXXXX.00 - that would be ridiculous.